core/services/vrf/listener_v2.go

```diff
@@ -426,7 +427,14 @@ func (lsn *listenerV2) shouldProcessSub(subID uint64, sub vrf_coordinator_v2.Get
 		return false
 	}
 
-	vrfRequest := reqs[0].req
+	// Sort requests in ascending order by CallbackGasLimit.
+	sortedReqs := make([]pendingRequest, len(reqs))
+	copy(sortedReqs, reqs)
+	sort.Slice(sortedReqs, func(a, b int) bool {
+		return sortedReqs[a].req.CallbackGasLimit < sortedReqs[b].req.CallbackGasLimit
+	})
```

Comment:
Lets use the generic version from exp/slices instead: https://pkg.go.dev/golang.org/x/exp/slices#SortFunc. I think we should have it available already as a dependency.